### DISCUSSION TITLE
Repository's go vanity url

### DISCUSSION THREAD

**Discussion by asimpleidea** [CloudNativeSDWAN/cnwan-operator#23] [Ideas]
Hey everyone,
I was thinking about something that I have been seeing commonly for other go projects around.
Every (modern) go project has a `mod` file that defines the dependencies that project needs. On top of this file you can see the name of the project as a repository, i.e. `github.com/CloudNativeSDWAN/cnwan-operator`.

You use this to pull the repository from other go file as a dependency, example:

```go
import (
reg "github.com/CloudNativeSDWAN/cnwan-operator/pkg/servregistry"
)
```

or from bash when downloading for local use, example:

```bash
go get github.com/CloudNativeSDWAN/cnwan-reader
```

Go supports what are called *vanity url*s, which are a way to call your repository as you want, even if the page does not really exist, just as long as it can point to an existing git repository, i.e. with a redirect. 
For example, Kubernetes has `k8s.io`, so when you load packages from go code you would do something like 

```go
import(
"k8s.io/api"
client "k8s.io/client-go"
)
```

or when you download it through shell:

```bash
go get k8s.io/api
```

even if the code is on github, or with etcd 

```go
import clientv3 "go.etcd.io/etcd/v3"
```

I didn't dig too much on the ways to do this exactly because it's not really the point of this topic.
The reason why I start this discussion is because I think it would be nice and better looking (and a bit more serious/professional?) if in future CN-WAN repositories were using vanity urls as well, like `go.cnwan.io/operator/service-registry` or `cnwan.cisco.com/reader`. One advantage about this, but it doesn't matter to us, is that if we decide to move the repositories somewhere else from github, then it would be just a matter of changing the redirection to point to the new website and whole thing would be totally transparent.
So yeah the whole thing is just a matter of vanity 😝

**Comment by ljakab** [comment]
We already have the `cnwan.io` domain configured to point to `github.com/CloudNativeSDWAN/`. I just tried, and if we add a suffix to `cnwan.io`, it gets appended to the GitHub URL. So I think this is a good idea, and we could already switch to using `cnwan.io/cnwan-operator` as the package name. Can you test with your personal repo if changing the URL the build still succeeds by pulling from the main repo?
